wallet/udb/vsp.go

```diff
@@ -26,15 +28,55 @@ const (
 	// paid, but not published.
 	VSPFeeProcessPaid
 	VSPFeeProcessErrored
+	// VSPFeeProcessConfirmed represents the state where the fee has been
+	// confirmed by the VSP.
+	VSPFeeProcessConfirmed
 )
 
 type VSPTicket struct {
 	FeeHash     chainhash.Hash
 	FeeTxStatus uint32
+	VSPHostID   uint32
+	Host        string
+	PubKey      []byte
 }
 
 // SetVSPTicket sets a VSPTicket record into the db.
 func SetVSPTicket(dbtx walletdb.ReadWriteTx, ticketHash *chainhash.Hash, record *VSPTicket) error {
+	// Check for Host/Pubkey buckets
+	pubkey, err := GetVSPPubKey(dbtx, []byte(record.Host))
+	if err != nil && errors.Is(err, errors.NotExist) {
+		// Pubkey entry doesn't exist for that entry, so create new one for host
+		vspHostBucket := dbtx.ReadWriteBucket(vspHostBucketKey)
+		v := vspHostBucket.Get(rootVSPHostIndex)
+		vspHostID := byteOrder.Uint32(v)
+		vspHostID += 1 // Bump current index
+		err = SetVSPHost(dbtx, vspHostID, &VSPHost{
+			Host: []byte(record.Host),
+		})
+		if err != nil {
+			return err
+		}
+
+		vspIndexBytes := make([]byte, 4)
+		byteOrder.PutUint32(vspIndexBytes, vspHostID)
+		err = vspHostBucket.Put(rootVSPHostIndex, vspIndexBytes)
+		if err != nil {
+			return errors.E(errors.IO, err)
+		}
+
+		err = SetVSPPubKey(dbtx, []byte(record.Host), &VSPPubKey{
+			ID:     vspHostID,
+			PubKey: record.PubKey,
+		})
+		if err != nil {
+			return err
+		}
+		record.VSPHostID = vspHostID
+	} else {
```

Comment:
You need to add a test for `err != nil` somewhere, otherwise this is also executed when the `GetVSPPubKey` failed for other reasons.